src/registrar/models/domain.py

```diff
@@ -91,221 +93,193 @@ class Status(models.TextChoices):
-                raise ValueError("Cannot activate. Application must be approved.")
-        if Domain.objects.filter(name=self.name, is_active=True).exists():
-            raise ValueError("Cannot activate. Domain name is already in use.")
-        # TODO: depending on the details of our registry integration
-        # we will either contact the registry and deploy the domain
-        # in this function OR we will verify that it has already been
-        # activated and reject this state transition if it has not
-        pass
+            {"registrant": "jd1234", "admin": "sh8013",...}
+        """
+        raise NotImplementedError()
+
+    @property
+    def creation_date(self) -> date:
+        """Get the `cr_date` element from the registry."""
+        raise NotImplementedError()
 
-    @transition(field="is_active", source="*", target=False)
-    def deactivate(self):
-        """This domain should not be live."""
-        # there are security concerns to having this function exist
-        # within the codebase; discuss these with the project lead
-        # if there is a feature request to implement this
-        raise Exception("Cannot revoke, contact registry.")
+    @property
+    def last_transferred_date(self) -> date:
+        """Get the `tr_date` element from the registry."""
+        raise NotImplementedError()
 
     @property
-    def sld(self):
-        """Get or set the second level domain string."""
-        return self.name.split(".")[0]
-
-    @sld.setter
-    def sld(self, value: str):
-        parts = self.name.split(".")
-        tld = parts[1] if len(parts) > 1 else ""
-        if Domain.string_could_be_domain(f"{value}.{tld}"):
-            self.name = f"{value}.{tld}"
-        else:
-            raise ValidationError("%s is not a valid second level domain" % value)
+    def last_updated_date(self) -> date:
+        """Get the `up_date` element from the registry."""
+        raise NotImplementedError()
 
     @property
-    def tld(self):
-        """Get or set the top level domain string."""
-        parts = self.name.split(".")
-        return parts[1] if len(parts) > 1 else ""
-
-    @tld.setter
-    def tld(self, value: str):
-        sld = self.name.split(".")[0]
-        if Domain.string_could_be_domain(f"{sld}.{value}"):
-            self.name = f"{sld}.{value}"
-        else:
-            raise ValidationError("%s is not a valid top level domain" % value)
+    def expiration_date(self) -> date:
+        """Get or set the `ex_date` element from the registry."""
+        raise NotImplementedError()
 
-    def __str__(self) -> str:
-        return self.name
+    @expiration_date.setter  # type: ignore
+    def expiration_date(self, ex_date: date):
+        raise NotImplementedError()
 
-    def nameservers(self) -> List[str]:
-        """A list of the nameservers for this domain.
+    @property
+    def password(self) -> str:
+        """Get the `auth_info.pw` element from the registry. Not a real password."""
```

Comment:
**question**: This comment about not being a real password is surprising enough that I feel like I want more explanation about what it means. Can we add more here, or maybe better, add a section in our documentation on how EPP uses `auth_info.pw` and then link to that section in comments like this?